### PR TITLE
Add tests, policy properties and move regex builder

### PR DESCRIPTION
**Pull request by jsoucheiron**

### CODE REVIEW

File: pycfmodel/utils.py [hunk 1]
```diff
@@ -14,3 +15,13 @@ def is_conditional_dict(value: Any) -> bool:
...
+
+
+def regex_from_cf_string(action: str) -> Pattern:
+    # Replace *
+    action = action.replace("*", ".*")
+
+    # Replace ?
+    action = action.replace("?", ".{1}")
+
+    return re.compile(f"^{action}$", re.IGNORECASE)
```
Comment (oscarbc96) [inline pycfmodel/utils.py:27]: What do you think about rewriting this function using fnmatch now that is moved? https://docs.python.org/3/library/fnmatch.html

Comment (jsoucheiron) [reply pycfmodel/utils.py:27]: Reading the docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_resource.html, it doesn't look like all fnmatch replacements are needed. Specially ranges with square brackets

File: pycfmodel/model/resources/iam_group.py [hunk 2]
```diff
@@ -35,4 +36,14 @@ class IAMGroup(Resource):
 
     TYPE_VALUE: ClassVar = "AWS::IAM::Group"
     Type: str = TYPE_VALUE
-    Properties: Resolvable[IAMGroupProperties]
+    Properties: Optional[Resolvable[IAMGroupProperties]]
+
+    @property
+    def policy_documents(self) -> List[OptionallyNamedPolicyDocument]:
+        result = []
+        if self.Properties.Policies:
```
Comment (ocrawford555) [inline pycfmodel/model/resources/iam_group.py:44]: Is this needed? Just asking as the `iam_role` resource file does not include this check and goes straight to the `for` loop.

Comment (jsoucheiron) [reply pycfmodel/model/resources/iam_group.py:44]: Policies is optional and it can be None

Comment (ocrawford555) [reply pycfmodel/model/resources/iam_group.py:44]: Cool - in that case shall we add this check to the `iam_role` resource? https://github.com/Skyscanner/pycfmodel/pull/51/files#diff-cdbcad10dd72e985485f15bd8ecfe4155a8e82923e12e8a829bb8c56d1b1ae34R50

Comment (jsoucheiron) [reply pycfmodel/model/resources/iam_group.py]: Good catch, I've fixed it.

**Review by ocrawford555 — APPROVED** [review]
LGTM 🚢

**Review by ocrawford555 — APPROVED** [review]

**Review by oscarbc96 — APPROVED** [review]
Very niceee 🚀

**Pull request merged**